
syntax = "proto3";

package protobuf;

message Network {   //Level 0

    message Layer {     //Level 1

        string type = 1;
        string name = 2;
        string input = 3;
        int32 Nn = 4;
        int32 Kx = 5;
        int32 Ky = 6;
        int32 stride = 7;
        int32 padding = 8;
        repeated int32 wgt_shape = 9 [packed=true]; //Remove headers
        repeated int32 act_shape = 10 [packed=true];
        repeated int32 out_act_shape = 11 [packed=true];

        // Not using Any for performance
        repeated float wgt_data_flt = 12 [packed=true];
        repeated float act_data_flt = 13 [packed=true];
        repeated float out_act_data_flt = 14 [packed=true];

    }

    string name = 1;
    repeated Layer layers = 2;

}